### PR TITLE
Singularity-3revert

### PR DESCRIPTION
**Pull request by megosato**
…63182e7c65a15e4a102087f to d019c3328d6395e33e35e5f2ddfc7059299238fd (inclusive) (#264)"

This reverts commit a0ebabe70b04c99ae01ba5070d1f4fdaaf4badfe.

### CODE REVIEW

File: ever_given/ever_given/engines/singularity_engine.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+from pathlib import Path
+import subprocess
+import typing
+import shlex
+import os
+
+from .utils import ContainerInstance, Engine, GUEST_OUTPUT_DIR
+
+AFTER_TERMINATE_WAIT = 5  # Seconds to wait after 'terminate' signal before 'kill'
+SINGULARITY_CONTAINER_TYPES = ["docker", "singularity", "singularity_sif"]
```
Comment (robbason) [inline ever_given/ever_given/engines/singularity_engine.py:11]:
```suggestion
SINGULARITY_CONTAINER_TYPES = ["docker", "singularity_remote", "singularity_local"]
```

File: ever_given/ever_given/tests/test_cancellation.py [hunk 2]
```diff
@@ -1,23 +1,41 @@
...
+    if container_engine == "singularity":
+        command = f"singularity pull docker://{container_uri}"
+        proc = subprocess.Popen(command, shell=True)
+        proc.wait()
+        container_uri = "logging-example_latest.sif"
+        container_type = "singularity_sif"
+        assert os.path.exists(container_uri)
+    if container_engine == "docker":
+        command = f"docker pull {container_uri}"
+        proc = subprocess.Popen(command, shell=True)
```
Comment (robbason) [inline ever_given/ever_given/tests/test_cancellation.py:24]: use subprocess.run here and above

**Review by robbason — CHANGES_REQUESTED** [review]

**Comment by megosato** [comment]
@robbason addressed your comments from here: https://github.com/samplchallenges/SAMPL-league/pull/278

I was using that branch to compare time if I downloaded all the sif files from ever_given once to see if it significantly improved testing time.

**Review by robbason — APPROVED** [review]

File: app/sampl/base_settings.py [hunk 3]
```diff
@@ -100,17 +101,50 @@
 
 CRISPY_TEMPLATE_PACK = "bootstrap4"
 
-LOGIN_TO_AWS = False
+# container engine is either "docker" or "singularity"
+CONTAINER_ENGINE = "docker"
```
Comment (megosato) [inline app/sampl/base_settings.py:105]: @robbason should this (CONTAINER_ENGINE) be pulled from environment variables?

Comment (robbason) [reply app/sampl/base_settings.py:105]: Let's discuss it in the context of the remote scheduler setup. In my opinion, we could go two ways:
1- environment variables all the way
2 - env variables for things that shouldn't be in config files, like hostnames and passwords, and use different django settings modules for settings that should be grouped together. For example, if you have a remote scheduler, container engine should be singularity.

**Pull request merged**